### DISCUSSION TITLE
Implementing crossnobis RSA when each run only has half of the conditions

### DISCUSSION THREAD

**Discussion by jimzhang629** [rsagroup/rsatoolbox#425] [Q&A]
Hi!

I am trying to do crossnobis RSA using rsatoolbox's methods on my fmri dataset. However, my experiment is a little convoluted and I'm not sure exactly how to do this using rsatoolbox's methods out of the box.

In my experiment, blocks can be one of four block types A,B,C,D. Each block type contains four unique conditions. Thus, I have 16 conditions total.

I have four runs. Each run contains two blocks, with the caveat that the two blocks in the same run will not be of the same block type, and also that block types A and C, and B and D are will not be in the same run. Which block types each run contains is randomly assigned per subject, with the constraint that each block type will appear in two runs across the experiment.

### Run 1
- **Block 1**: Block Type A (contains four unique conditions)
- **Block 2**: Block Type B (contains four unique conditions)

### Run 2
- **Block 1**: Block Type D (contains four unique conditions)
- **Block 2**: Block Type C (contains four unique conditions)

### Run 3
- **Block 1**: Block Type D (contains four unique conditions)
- **Block 2**: Block Type A (contains four unique conditions)

### Run 4
- **Block 1**: Block Type C (contains four unique conditions)
- **Block 2**: Block Type B (contains four unique conditions)

### Summary of constraints
- **Block Types**:
  - A, B, C, D, each containing 4 unique conditions.
- **Run Composition**:
  - Each run contains two different block types.
  - Block types A and C and B and D do not appear together in the same run.
- **Random Assignment**:
  - Block types are randomly assigned to runs with the constraint that each block type appears in two runs.
  
Basically, what this means is that in each run, only 8 of the 16 conditions that I want to compare in my RSA are represented. I looked into creating [partial RDMs](https://rsatoolbox.readthedocs.io/en/stable/rescale_partials.html) for each run and then averaging, but I'm not sure how this would work in a crossnobis implementation where I want to cross-validate across runs.

Any advice would be greatly appreciated :)

**Comment by JasperVanDenBosch** [comment]
Hi @jimzhang629 ,
With this design, I can't think of a way to create crossvalidated RDM's on the participant level without breaking the typical assumptions. But as long as you do mean removal I think you can make non-crossvalidated participant RDMs that might work fairly well

**Reply by jimzhang629** [reply]
Thanks so much for the quick reply, Jasper :)

Another problem that I should have mentioned is that my trial counts between the four conditions in each block are imbalanced. Specifically, the ratio of trial counts is 1:3:3:9 between the four conditions in each block (though I only use correct trials, so in reality, could be even worse). I have 64 trials per block, so the least common condition only appears in 4 trials and the most common condition appears in 36 trials. This is another reason I was trying to do the crossnobis RSA, to help correct for this trial count imbalance.

Thus, I have some more questions on how exactly to do the mean-centered RSA implementation.

1. When calculating the run mean, should I weight each condition's beta weights by the number of correct trials it had in the run?
2. After centering my beta weights for each condition using this weighted run mean, I think I should then divide these run mean-centered beta weights by their respective standard errors to standardize the variance across conditions. Is this correct?
3. And then I average these mean-centered, standardized beta weights for each condition across the two runs it appears. But when computing this average, again weigh the beta weights from each run by the number of (correct) trials they were calculated from. Does this sound right..?
4. And finally, use some distance metric (probably Euclidean is best?) between these averaged, mean-centered, standardized beta weights to compute my neural RDM.

Again, thanks in advance for any advice on the implementation!

**Reply by JasperVanDenBosch** [reply]
For some reason real life experiments never seem to neatly fit the ideal :)
I don't see anything obviously wrong with what you're describing but it's fairly complicated.
I have an idea for something that could work; basically like [Alink 2015](https://scholar.google.co.uk/citations?view_op=view_citation&hl=en&user=XQ3-JHgAAAAJ&citation_for_view=XQ3-JHgAAAAJ:mvPsJ3kp5DgC) you could instead account for the variance caused by the runs in the first level analysis with a run-predictor. May be easier to talk on a video call?

**Reply by jimzhang629** [reply]
Haha yes having an experiment that fit the ideal would be nice!

Regarding the Alink 2015 approach -  I assume that I'd add four run-predictors at the subject level, one for each run, such that each run-predictor would be 1 for conditions from its corresponding run and 0 otherwise? However, at the run level in FSL, I've already added a couple of block order nuisance regressors in my run-level model, such that one regressor is 1 for all of block 1 and 0 for all of block 2, and the other is the opposite. I'm not sure if this already accounts for run-specific variance or not, and also if this complicates things.

I'm also not sure if trying the cross-run RSA approach that was also mentioned in Alink 2015 would work too. In this case, I think I'd find the dissimilarity between conditions from different runs, and then average each condition pair across runs. This seems a bit tricky to do though since conditions from the same block type could only be compared between two runs, while conditions from two different block types could be compared between more runs.

Basically, I don't really know what I'm doing and think a video call would be super helpful if you have time :)
My email is jzhang7345@gmail.com if you want to coordinate over email.
